packages/pwa-kit-react-sdk/src/ssr/universal/hooks/use-get-props-args.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useContext} from 'react'
+import {useParams, useLocation} from 'react-router-dom'
+import AppConfig from '../../universal/components/_app-config'
+
+export interface GetPropsArgs {
+    location: Location
+    params: {[key: string]: string}
+    req: object | undefined // Get type from runtime?
+    res: object | undefined // Get type from runtime?
+    [key: string]: object | string | number // This accounts for all the `extraGetPropsArgs` values.
```

Comment:
```suggestion
    [key: string]: object | string | number | undefined // This accounts for all the `extraGetPropsArgs` values.
```
Need `undefined` in the index signature if you have optional properties. 😉